File: 1963. Minimum Number of Swaps to Make the String Balanced.java
// 1963. Minimum Number of Swaps to Make the String Balanced
//s = "][]["
class Solution {
    public int minSwaps(String s) {
    int min=0;
    for(int i=0;i<s.length();i++){
        if(s.charAt(i)=='[') min++;
        else if(min>0)min--;
    }return (min+1)/2;
    }
}

File: 1963. Minimum Number of Swaps to Make the String Balanced.js
// 1963. Minimum Number of Swaps to Make the String Balanced
s = "][]["
/**
 * @param {string} s
 * @return {number}
 */
//SOL 1 | CORRECTED BY Math.floor()
var minSwaps = function(s) {
    var open_bracket_count = 0
    Array.from(s).forEach(char => {
        if (char == '[') {
            open_bracket_count += 1
        } else if(open_bracket_count>0){
          open_bracket_count-=1
        }
    })
    return Math.floor((open_bracket_count+ 1 ) / 2)
};
console.log(minSwaps(s))

